contracts/peg/curve/CurveRouter.sol

```diff
@@ -0,0 +1,340 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.4;
+
+import {ICurveRouter} from "./ICurveRouter.sol";
+import {ICurvePool} from "./ICurvePool.sol";
+import {IVolt} from "../../volt/IVolt.sol";
+import {IPegStabilityModule} from "../IPegStabilityModule.sol";
+import {SafeERC20, IERC20} from "@openzeppelin/contracts/token/ERC20/utils/SafeERC20.sol";
+import {SafeCast} from "@openzeppelin/contracts/utils/math/SafeCast.sol";
+import {MainnetAddresses} from "../../test/integration/fixtures/MainnetAddresses.sol";
+import {Constants} from "../../Constants.sol";
+
+contract CurveRouter is ICurveRouter {
+    using SafeERC20 for IERC20;
+    using SafeCast for *;
+
+    /// @notice reference to the Volt contract used.
+    /// Router can be redeployed if Volt address changes
+    IVolt public immutable override volt;
+
+    constructor(IVolt _volt, CurveApproval[5] memory curveApprovals) {
+        volt = _volt;
+
+        unchecked {
+            for (uint256 i = 0; i < curveApprovals.length; i++) {
+                IERC20(curveApprovals[i].token).safeApprove(
+                    curveApprovals[i].pool,
+                    type(uint256).max
+                );
+            }
+        }
+
+        IERC20(MainnetAddresses.VOLT).approve(
+            address(MainnetAddresses.VOLT_USDC_PSM),
+            type(uint256).max
+        );
+
+        IERC20(MainnetAddresses.USDC).approve(
+            address(MainnetAddresses.VOLT_USDC_PSM),
+            type(uint256).max
+        );
+
+        IERC20(MainnetAddresses.FEI).approve(
+            address(MainnetAddresses.VOLT_FEI_PSM),
+            type(uint256).max
+        );
+
+        IERC20(MainnetAddresses.VOLT).approve(
+            address(MainnetAddresses.VOLT_FEI_PSM),
+            type(uint256).max
+        );
```

Comment:
we should just make this a loop just like the curve approvals so we aren't hardcoding these addresses in the constructor, which will make L2 deployment require code changes.